heat/core/dndarray.py

```diff
@@ -730,6 +730,32 @@ def __getitem__(self, key: Union[int, Tuple[int, ...], List[int, ...]]) -> DNDar
             slices = [slice(None)] * (self.ndim - (len(kst) + len(kend)))
             key = kst + slices + kend
 
+        def __is_singular(obj: any) -> bool:
```

Comment:
I'm sorry and don't want to cause more work, but honestly we already have the `self_proxy` at this point.
Something like the following could be 100 % torch-compatible (i've got no idea about performance or edge cases though)
```python
zeros = tuple([0] * (self.ndim -1))
is_singular = bool(1 - self_proxy[ *zeros[:i], key[i], *zeros[i:] ].ndim)
```